cassandra/src/main/java/io/micronaut/configuration/cassandra/health/CassandraHealthIndicator.java

```diff
@@ -0,0 +1,154 @@
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.cassandra.health;
+
+import com.datastax.oss.driver.api.core.AllNodesFailedException;
+import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.*;
+
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.metadata.NodeState;
+import com.datastax.oss.driver.api.core.servererrors.QueryExecutionException;
+import com.datastax.oss.driver.api.core.servererrors.QueryValidationException;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.health.HealthStatus;
+import io.micronaut.management.endpoint.health.HealthEndpoint;
+import io.micronaut.management.health.indicator.AbstractHealthIndicator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import java.net.InetSocketAddress;
+import java.util.*;
+
+/**
+ * A {@link io.micronaut.management.health.indicator.HealthIndicator} for Cassandra.
+ *
+ * @author Ilkin Ashrafli
+ * @since 2.1.0
+ */
+@Requires(property = HealthEndpoint.PREFIX + ".cassandra.enabled", notEquals = "false")
+@Requires(beans = HealthEndpoint.class)
+@Singleton
+public class CassandraHealthIndicator extends AbstractHealthIndicator<Map<String, Object>> {
+
+    private static final String COL_RELEASE_VERSION = "release_version";
+    private static final String COL_CLUSTER_NAME = "cluster_name";
+    private static final String COL_CQL_VERSION = "cql_version";
+    private static final String COL_DATA_CENTER = "data_center";
+    private static final String COL_NATIVE_PROTOCOL_VERSION = "native_protocol_version";
+    private static final String COL_PARTITIONER = "partitioner";
+    private static final String COL_RACK = "rack";
+
+    private static final SimpleStatement VALIDATION_SELECT = selectFrom("system", "local")
+            .column(COL_RELEASE_VERSION)
+            .column(COL_CLUSTER_NAME)
+            .column(COL_CQL_VERSION)
+            .column(COL_DATA_CENTER)
+            .column(COL_NATIVE_PROTOCOL_VERSION)
+            .column(COL_PARTITIONER)
+            .column(COL_RACK)
+            .build()
+            .setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
```

Comment:
```suggestion
            .setConsistencyLevel(ConsistencyLevel.LOCAL_ONE)
            .setTimeout(Duration.ofSeconds(10));
```

If you really want to keep this query, I suggest hard-coding the timeout to something comfortable.